lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -374,9 +374,25 @@ def full_replacement_map
                          @define_variables)
     end
 
+    def read_dynamic_device(device)
+      file_full_path = File.realpath("#{DEVICES_JSON_DIR}/#{device}.rb")
+      require file_full_path
+
+      raise InvalidConfigFile, "#{device} does not exist" unless QemuHCK::Devices.respond_to?(device)
+
+      device_obj = QemuHCK::Devices.public_send(device, @logger)
```

Comment:
Let's call it `info` as done elsewhere. `obj` is meaningless as everything in Ruby is an object and shouldn't be used for names.